cluster/job_once_example_test.go

```diff
@@ -0,0 +1,40 @@
+package cluster
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+func HandleJobOnceCalls(key string) {
+	if key == "the key i'm watching for" {
+		// Work to do only once per cluster
+	}
+}
+
+func ExampleScheduleOnce() {
+	// Use p.API from your plugin instead.
+	pluginAPI := plugin.API(nil)
+
+	// Get the scheduler, which you can pass throughout the plugin...
+	scheduler := GetJobOnceScheduler(pluginAPI)
+
+	// Set the plugin's callback handler
+	_ = scheduler.SetCallback(HandleJobOnceCalls)
+
+	// Now start the scheduler, which starts the poller and schedules all waiting jobs.
+	_ = scheduler.Start()
+
+	// main thread...
+
+	// add a job
+	_, _ = scheduler.ScheduleOnce("the key i'm watching for", time.Now().Add(2*time.Hour))
```

Comment:
The nomenclature `scheduler.ScheduleOnce` feels like a mouthful, especially since it's not likely we'll want to extend this particular API to `ScheduleMultipleTimes` or `ScheduleRecurring` or some such. I wonder if we could simplify the nomenclature and take inspiration from the Unix command line tool `at` to do something like:

```go
pluginAPI := plugin.API(nil)
callback := func(key string) {}

once := NewScheduler(pluginAPI, callback)

scheduler.At("key", futureTime)

// extend API surface trivially?
scheduler.In("key", 2*time.Hour)

// do I have to close it?
defer once.Close()
```

Reply:
Sure, I'm open to changing the verb. But I'm not sure I love `scheduler.At`--we can't control what the client calls the var. maybe `x.ScheduleAt`? Then `y.ScheduleIn` works too. 

Also, the code above would be:
```
pluginAPI := plugin.API(nil)
callback := func(key string) {}

once := NewScheduler(pluginAPI, callback)

job, _ := once.At("key", futureTime)

// extend API surface trivially?
once.In("key", 2*time.Hour)

// do I have to close it?
defer job.Close()
```

(No need to call `defer job.Close()` -- that would cancel the scheduled job it right away)

Reply:
I'm 0/5 on the `ScheduleOnce` name, but it's true that the `Close` verb confused me at the beginning, thinking that I needed to call that too. We could change it to `Cancel`, `Stop`, `Kill` or something more explicit?